cli/tests/test_duplicate_command.rs

```diff
@@ -2427,6 +2427,118 @@ fn test_rebase_duplicates() {
     ");
 }
 
+#[test]
+fn test_duplicate_description_template() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+
+    create_commit(&work_dir, "a", &[]);
+    create_commit(&work_dir, "b", &["a"]);
+    create_commit(&work_dir, "c", &["b"]);
+
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  7e4fbf4f2759   c
+    ○  1394f625cbbd   b
+    ○  2443ea76b0b1   a
+    ◆  000000000000
+    [EOF]
+    ");
+
+    // Test duplicate_commits()
+    test_env.add_config(r#"templates.duplicate_description = "concat(description, '\n(cherry picked from commit ', commit_id, ')')""#);
+    let output = work_dir.run_jj(["duplicate", "a"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Duplicated 2443ea76b0b1 as yostqsxw 20d8e008 a
+    [EOF]
+    ");
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  7e4fbf4f2759   c
+    ○  1394f625cbbd   b
+    ○  2443ea76b0b1   a
+    │ ○  20d8e008cc0f   a
+    ├─╯
+    ◆  000000000000
+    [EOF]
+    ");
+
+    // Test duplicate_commits_onto_parents()
+    let output = work_dir.run_jj(["duplicate", "a", "-B", "b"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Warning: Duplicating commit 2443ea76b0b1 as a descendant of itself
+    Duplicated 2443ea76b0b1 as kpqxywon cdddb77e (empty) a
+    Rebased 2 commits onto duplicated commits
+    Working copy  (@) now at: royxmykx 3fb5c9d2 c | c
+    Parent commit (@-)      : zsuskuln 2ac3bd84 b | b
+    [EOF]
+    ");
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  3fb5c9d2839f   c
+    ○  2ac3bd84343b   b
+    ○  cdddb77efea3   a
+    ○  2443ea76b0b1   a
+    │ ○  20d8e008cc0f   a
+    ├─╯
+    ◆  000000000000
+    [EOF]
+    ");
+
+    // Test empty template
+    test_env.add_config("templates.duplicate_description = ''");
+    let output = work_dir.run_jj(["duplicate", "b", "-d", "root()"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Duplicated 2ac3bd84343b as wqnwkozp 6537441b (no description set)
+    [EOF]
+    ");
+
+    // Test `description` as an alias
+    test_env.add_config("templates.duplicate_description = 'description'");
+    let output = work_dir.run_jj([
+        "duplicate",
+        "c",
+        "-d",
+        "root()",
+        // Use an argument here so we can actually see the log in the last test
+        // (We don't have a way to remove a config in TestEnvironment)
+        "--config",
+        "template-aliases.description='\"alias\"'",
+    ]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Duplicated 3fb5c9d2839f as lylxulpl e6877e1e alias
+    [EOF]
+    ");
+
+    let template = r#"commit_id.short() ++ "\n" ++ description"#;
+    let output = work_dir.run_jj(["log", "-T", template]);
```

Comment:
@yuja Do you think it's worthwhile to run `jj log` at every other commit if we're doing a big one at the end? Probably would be best to remove the intermediate ones.

Reply:
Yeah, I would also remove intermediate `jj log` calls so long as the result is obvious.

Reply:
> so long as the result is obvious

Do you think that's true in this case? :)